src/NServiceBus.Core/Performance/ReceiveStatisticsFeature.cs

```diff
@@ -14,9 +14,19 @@ public ReceiveStatisticsFeature()
         }
         protected internal override void Setup(FeatureConfigurationContext context)
         {
-            context.Pipeline.Register("ReceivePerformanceDiagnosticsBehavior", typeof(ReceivePerformanceDiagnosticsBehavior), "Provides various performance counters for receive statistics");
+            var transportAddress = context.Settings.LocalAddress();
+
+            context.Container.RegisterSingleton(new ReceivePerformanceDiagnosticsBehavior(transportAddress));
```

Comment:
with the current implementation that shouldn't make a difference I think? But InstancePerCall is probably better, I agree.